### PR TITLE
[service.libraryautoupdate] 1.0.3

### PR DESCRIPTION
**Pull request by robweber**
### Description
Updating the Library Auto Updater plugin for Krypton+ versions of Kodi. Specific updates in this release: 

* gracefully cancel out of custom dialog box
* replace builtin with json commands for executing scans
* catch cron syntax errors and inform the user
* added option to check sources before scan

### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by robweber** [comment]
It looks like the failed checks are python 2.7 vs 3 syntax errors. Do I need to fix those now or is that just guidance for going forward when python 3 is enforced?

**Comment by Rechi** [comment]
The raise and repr syntax changes are enforced (reported as error). They would be even compatible with Python 2.6.
Is there any particular reason to not update the version in jarvis branch?

**Comment by robweber** [comment]
I'll fix those up, weird they haven't been issues before but I haven't done a full repo pull in a while. 

As far as updating previous Kodi versions. I'd have to check specifically if something is dependent on a particular library. I try and maintain 1 version behind the most latest Kodi version if I can. I know some minor things, like tags in the addon.xml file, are Krypton specific. I suppose it's just more a practice of not trying to go too far back. I figured at this point Krypton+ is probably the version most users are on so I'd rather stay current.

File: service.libraryautoupdate/resources/lib/utils.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    if(not xbmcvfs.exists(xbmc.translatePath(data_dir()))):
+        xbmcvfs.mkdir(xbmc.translatePath(data_dir()))
+
+def data_dir():
+    return __Addon.getAddonInfo('profile')
+
+def addon_dir():
+    return __Addon.getAddonInfo('path')
+
+def log(message,loglevel=xbmc.LOGNOTICE):
```
Comment (Rechi) [inline service.libraryautoupdate/resources/lib/utils.py]: Please change this default to debuging and only log the messages really required with lognotice.
```suggestion
def log(message,loglevel=xbmc.LOGDEBUG):
```

**Comment by Rechi** [comment]
From a quick look only icon path and the updated xbmc.python version would be incompatible with Jarvis. I just asked as I want to avoid having outdated version about which the maintainer doesn't care anymore everywhere.

>  Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Please squash your commits.

**Comment by robweber** [comment]
I tried squashing it all down into one commit and am pretty sure I hosed the whole thing up. Going to close this and start over.

**Pull request closed without merging**